core/src/main/java/be/orbinson/aem/dictionarytranslator/services/DictionaryService.java

```diff
@@ -33,11 +46,52 @@ public interface DictionaryService {
 
     boolean keyExists(Resource dictionaryResource, String language, String key);
 
+    /**
+     * 
+     * @param languageResource
+     * @param key
+     * @return
+     * @deprecated use {@link #getMessages(Resource, String)} instead
+     */
+    @Deprecated
     Resource getMessageEntryResource(Resource languageResource, String key);
 
+    @Deprecated
+    /**
+     * 
+     * @param resourceResolver
+     * @param dictionaryResource
+     * @param language
+     * @param key
+     * @param message
+     * @throws PersistenceException
+     * @deprecated Use {@link #updateMessageEntry(ResourceResolver, Resource, String, String, String)} instead
+     */
     void createMessageEntry(ResourceResolver resourceResolver, Resource dictionaryResource, String language, String key, String message) throws PersistenceException;
 
     void updateMessageEntry(ResourceResolver resourceResolver, Resource dictionaryResource, String language, String key, String message) throws PersistenceException, RepositoryException;
 
     void deleteMessageEntry(ResourceResolver resourceResolver, Resource combiningMessageEntryResource) throws PersistenceException, ReplicationException;
+
+    Map<String, String> getMessages(Resource dictionaryResource, String language);
+    
+    enum DictionaryType {
+        SLING_MESSAGE_ENTRY,
+        JSON_FILE,
+        /** the languages below this dictionary have different types, only applicable on dictionary level, not on language level */
+        MIXED
```

Comment:
What happens with the edit ability / replication if it is of typed mixed?

Reply:
read-only